homeassistant/components/blueprint/importer.py

```diff
@@ -245,13 +245,32 @@ async def fetch_blueprint_from_website_url(
     return ImportedBlueprint(suggested_filename, raw_yaml, blueprint)
 
 
+async def fetch_blueprint_from_generic_url(
+    hass: HomeAssistant, url: str
+) -> ImportedBlueprint:
+    """Get a blueprint from a generic website."""
+    session = aiohttp_client.async_get_clientsession(hass)
+
+    resp = await session.get(url, raise_for_status=True)
+    raw_yaml = await resp.text()
+    data = yaml.parse_yaml(raw_yaml)
+
+    assert isinstance(data, dict)
+    blueprint = Blueprint(data)
+
+    parsed_import_url = yarl.URL(url)
+    suggested_filename = f"generic/{parsed_import_url.parts[-1][:-5]}"
```

Comment:
This is not a great folder name, can we instead use the domain name as folder name?
Also, if the use case you really want is gitlab, consider adding a specific importer for gitlab.

Reply:
:+1: Updated to use the URL host + rebased.

My specific use case is not GitLab, personally I'd like to use sr.ht. I also know of some people who might be interested in using Codeberg. Therefore, I think adding specific importers for each and every source forge might become a bit cumbersome, esp. in cases where people would like to use their personal homepage or blog.

Reply:
Yes, sure, the generic downloader added in this PR is fine 👍 
It's just that the specific downloaders can be a little bit smarter.